File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        val_idx = {}

        for i, n in enumerate(nums):
            wanted = target - n
            if wanted in val_idx and val_idx[wanted] != i:
                return [val_idx[wanted], i]
            else:
                val_idx[n]= i
        

File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> seen;
        
        for(int i=0; i < nums.size(); i++){
            if(seen.contains(nums[i])){
                return true;
            }else{
                seen[nums[i]] = 0;
            }
        }
        return false;
    }
};